posthog/hogql_queries/insights/trends/query_builder.py

```diff
@@ -0,0 +1,232 @@
+                    groupArray(count) AS total
+                FROM {inner_query}
+            """,
+            placeholders={"inner_query": inner_query},
+        )
+
+        if self._breakdown.enabled:
+            query.select.append(ast.Field(chain=["breakdown_value"]))
+            query.group_by = [ast.Field(chain=["breakdown_value"])]
+            query.order_by = [ast.OrderExpr(expr=ast.Field(chain=["breakdown_value"]), order="ASC")]
+
+        return query
+
+    def _inner_select_query(self, inner_query: ast.SelectUnionQuery) -> ast.SelectQuery:
+        query = parse_select(
+            """
+                SELECT
+                    sum(total) AS count,
+                    day_start
+                FROM {inner_query}
+                GROUP BY day_start
+                ORDER BY day_start ASC
+            """,
+            placeholders={"inner_query": inner_query},
+        )
+
+        if self._breakdown.enabled:
+            query.select.append(ast.Field(chain=["breakdown_value"]))
+            query.group_by.append(ast.Field(chain=["breakdown_value"]))
+            query.order_by.append(ast.OrderExpr(expr=ast.Field(chain=["breakdown_value"]), order="ASC"))
+
+        return query
+
+    def _events_filter(self) -> ast.Expr:
+        series = self.series
+        filters: List[ast.Expr] = []
+
+        # Dates
+        filters.extend(
+            [
+                parse_expr(
+                    "(toTimeZone(timestamp, 'UTC') >= {date_from})",
+                    placeholders=self.query_date_range.to_placeholders(),
+                ),
+                parse_expr(
+                    "(toTimeZone(timestamp, 'UTC') <= {date_to})",
+                    placeholders=self.query_date_range.to_placeholders(),
+                ),
+            ]
+        )
+
+        # Series
+        if series_event_name(self.series) is not None:
+            filters.append(
+                parse_expr(
+                    "event = {event}", placeholders={"event": ast.Constant(value=series_event_name(self.series))}
+                )
+            )
+
+        # Filter Test Accounts
+        if (
+            self.query.filterTestAccounts
+            and isinstance(self.team.test_account_filters, list)
+            and len(self.team.test_account_filters) > 0
+        ):
+            for property in self.team.test_account_filters:
+                filters.append(property_to_expr(property, self.team))
+
+        # Properties
+        if self.query.properties is not None and self.query.properties != []:
+            filters.append(property_to_expr(self.query.properties, self.team))
+
+        # Series Filters
+        if series.properties is not None and series.properties != []:
+            filters.append(property_to_expr(series.properties, self.team))
+
+        # Breakdown
+        if self._breakdown.enabled and not self._breakdown.is_histogram_breakdown:
+            filters.append(self._breakdown.events_where_filter())
```

Comment:
Hm... won't this just exclude everything that's not under the top N breakdown options? Is this how the current query worked? I'd assume we'd like everything else to be returned under a "other" blob? 🤔

Reply:
The current implementation only takes the top 25 results, ordered by `count()` at the moment. No bucketing via an "other" blob